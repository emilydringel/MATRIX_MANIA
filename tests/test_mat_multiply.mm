def matrix<int> multiply(matrix<int> x, matrix<int> y){
/* matrix<int> empty as parameter? */

    int sizeOfR1 = getRows(x);
    int sizeOfC1 = getColumns(x);
    int sizeOfR2 = getRows(y);
    int sizeOfC2= getColumns(y);

    matrix<int> empty = [0,0];

    for(int i = 0; i < sizeOfR1; i+1){
	for(int j = 0; j < sizeOfC2; j+1){
	    for(int k = 0; k < sizeOfC1; k+1){
		empty[i,j]+= x[i,k]*y[k,j];
	    }
  	}
    }
  return empty;
}

def int main(){
  
matrix<int> m = [1,4];
matrix<int> n = [2,3];

matrix<int> multiplied = multiply(m,n);

  printm(m);
  printm(n);
  printm(multiplied);
  printm(multiply([6,8],[9,10]));
}
