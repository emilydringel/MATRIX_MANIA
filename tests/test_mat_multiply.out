1 4
2 3
14
134